spec/operators/share-spec.ts

```diff
@@ -1,331 +1,466 @@
+    it('should be repeatable', () => {
+      const source =     cold('-1-2-3----4-|                        ');
+      const sourceSubs =     ['^           !                        ',
+                              '            ^           !            ',
+                              '                        ^           !'];
+      const shared = source.pipe(share());
+      const subscribe1 =      's                                    ';
+      const expected1 =       '-1-2-3----4--1-2-3----4--1-2-3----4-|';
+      const subscribe2 =      '    s                                ';
+      const expected2 =       '    -3----4--1-2-3----4--1-2-3----4-|';
+
+      expectObservable(hot(subscribe1).pipe(tap(() => {
+        expectObservable(shared.pipe(repeat(3))).toBe(expected1);
+      }))).toBe(subscribe1);
+
+      expectObservable(hot(subscribe2).pipe(tap(() => {
+        expectObservable(shared.pipe(repeat(3))).toBe(expected2);
+      }))).toBe(subscribe2);
+
+      expectSubscriptions(source.subscriptions).toBe(sourceSubs);
+    });
 
-    expectObservable(e1.pipe(share())).toBe(expected);
-  });
+    it('should not change the output of the observable when never', () => {
+      const e1 = NEVER;
+      const expected = '-';
+
+      expectObservable(e1.pipe(share())).toBe(expected);
+    });
+
+    it('should not change the output of the observable when empty', () => {
+      const e1 = EMPTY;
+      const expected = '|';
 
-  it('should not change the output of the observable when empty', () => {
-    const e1 = EMPTY;
-    const expected = '|';
+      expectObservable(e1.pipe(share())).toBe(expected);
+    });
 
-    expectObservable(e1.pipe(share())).toBe(expected);
+    // TODO: fix firehose unsubscription
+    it.skip('should stop listening to a synchronous observable when unsubscribed', () => {
+      const sideEffects: number[] = [];
+      const synchronousObservable = new Observable<number>(subscriber => {
+        // This will check to see if the subscriber was closed on each loop
+        // when the unsubscribe hits (from the `take`), it should be closed
+        for (let i = 0; !subscriber.closed && i < 10; i++) {
+          sideEffects.push(i);
+          subscriber.next(i);
+        }
+      });
+
+      synchronousObservable.pipe(
+        share(),
+        take(3),
+      ).subscribe(() => { /* noop */ });
+
+      expect(sideEffects).to.deep.equal([0, 1, 2]);
+    });
   });
 
-  // TODO: fix firehose unsubscription
-  it.skip('should stop listening to a synchronous observable when unsubscribed', () => {
-    const sideEffects: number[] = [];
-    const synchronousObservable = new Observable<number>(subscriber => {
-      // This will check to see if the subscriber was closed on each loop
-      // when the unsubscribe hits (from the `take`), it should be closed
-      for (let i = 0; !subscriber.closed && i < 10; i++) {
-        sideEffects.push(i);
-        subscriber.next(i);
-      }
+  describe('share(config)', () => {
```

Comment:
nitpick: In a previous review I mentioned that the parameter type is named `ShareOptions` whereas elsewhere type names like `XxxConfig` is used. Can we rename it to config? 🙏 I mean, it's referred to here as  config - not options. I am a fan of consistency, however trivial.